### ISSUE TITLE
Where does Perspective save edited images?

### ISSUE THREAD

**Issue by yephny** [danopdev/Perspective#3]
Where can I find the images edited by Perspective?

**Comment by danopdev** [comment]
They are saved in "Pictures\Perspective" but you should be able to find them in the Gallery application.
Unfortunately with the latest android versions my method to add them to Gallery does not works.
I change the method but it will be for the next release.

Can you please try with a file explorer if the files are present in "Pictures\Perspective" folder ?

**Comment by yephny** [comment]
That's where I checked as well, there's no `/sdcard/Pictures/Perspective`.

Android 10, LineageOS.

**Comment by danopdev** [comment]
It's strange the "sdcard" in the path.
The full path is actually "/storage/emulated/0/Pictures/Perspective" but maybe "/storage/emulated/0/" don't exists on all devices.

**Comment by yephny** [comment]
@danopdev On my device, `/sdcard/` just redirects to `/storage/emulated/0/`, which does exist.

**Comment by danopdev** [comment]
It's strange. I'll try to find another method to get Pictures folder.
Do you know the path on your device ?

**Comment by yephny** [comment]
It's `/storage/emulated/0/Pictures/`.

**Comment by yephny** [comment]
Comparing yours to [PixaPencil's code](https://github.com/Catrobat/Paintroid/blob/3253d71a55ea92cf22f7109701b634765407f293/Paintroid/src/main/java/org/catrobat/paintroid/common/Constants.kt#L63),

Perspective: ```Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_PICTURES).absolutePath```

PixaPencil: ```File(Environment.getExternalStorageDirectory(), Environment.DIRECTORY_PICTURES)```

Could it be because yours has the extra `Public` bit? Might also be `.absolutepath`.

**Comment by danopdev** [comment]
Actually Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_PICTURES).absolutePath is the current code, but previous release used a hardcoded location.

On my device both methods returns: /storage/emulated/0/Pictures

Can you see please check what are the values for both methods ?

**Comment by yephny** [comment]
@danopdev How do I check that?

**Comment by yephny** [comment]
It still won't work on version 1.8.

**Comment by danopdev** [comment]
You need to write a small program for that, but I can try a different approach:

If I generate 2 special builds:
1. File(Environment.getExternalStorageDirectory(), Environment.DIRECTORY_PICTURES) + "\Perspective"
2. File(Environment.getExternalStorageDirectory(), Environment.DIRECTORY_PICTURES)

Can you please test them ?

**Comment by yephny** [comment]
Sure, I can test them.

**Comment by danopdev** [comment]
In the ZIP file you can find the 2 applications:
* 1.10 = > try to save to File(Environment.getExternalStorageDirectory(), Environment.DIRECTORY_PICTURES)
* 1.12 => try to save to File(Environment.getExternalStorageDirectory(), Environment.DIRECTORY_PICTURES) + "\Perspective"

Please try both of them (I have a preference for 1.12)

[Test applications.zip](https://github.com/danopdev/Perspective/files/10446921/Test.applications.zip)

**Comment by yephny** [comment]
Tried both out, none works. Kotatsu, Imagepipe, PxerStudio, Resplash all work for reference.

**Comment by danopdev** [comment]
This is really strange.
Do you have the little toast saying the save was OK or failed ?

**Comment by yephny** [comment]
Yeah, the toast's there. It says `Saved to image_name.png`.

**Comment by danopdev** [comment]
OK, I didn't checked the save result.
This is a new build that do it and also show the full path in the toast.
Can you please try it ?
Also can you try to save it as JPEG too ?
[Perspective.zip](https://github.com/danopdev/Perspective/files/10464850/Perspective.zip)

**Comment by yephny** [comment]
I tried all the outputs, they all result in the same error: `Failed to save: /storage/emulated/0/Pictures/Perspective/image_name.jpeg`. Manually creating the Perspective directory won't help either.

**Comment by danopdev** [comment]
At least it's consistent with the result :)
What if you manually create the folders "Pictures/Perspective" ?
If still don't works, I'm out of ideas.

**Comment by yephny** [comment]
Nope, still won't work even with `Pictures/Perspective/` present.

**Comment by danopdev** [comment]
The only thing left is that I use OpenCV to write the file (it support TIFF & 16 bit PNG).
But this is not that important.
I will try to use android system to save the file.

**Comment by danopdev** [comment]
Last chance. This time I try to save the file using android libraries and not OpenCV.
Can you please give it a try ? 
[Perspective_1.14.apk.zip](https://github.com/danopdev/Perspective/files/10478533/Perspective_1.14.apk.zip)

**Comment by yephny** [comment]
@danopdev Once again, it fails.
